mrmustard/lab_dev/states/states.py

```diff
@@ -0,0 +1,135 @@
+
+        >>> state = Vacuum([1, 2])
+        >>> assert state.modes == [1, 2]
+
+    Args:
+        modes: A list of modes.
+
+    .. details::
+
+        The :math:`N`-mode vacuum state is defined by
+
+        .. math::
+            V = \frac{\hbar}{2}I_N \text{and } r = \bar{0}_N.
+
+        Its ``(A,b,c)`` triple is given by
+
+        .. math::
+            A = O_{N\text{x}N}\text{, }b = O_N\text{, and }c = 1.
+    """
+
+    def __init__(
+        self,
+        modes: Iterable[int],
+    ) -> None:
+        super().__init__("Vacuum", modes=modes)
+
+    @property
+    def representation(self) -> Bargmann:
+        num_modes = len(self.modes)
+        return Bargmann(*triples.vacuum_state_Abc(num_modes))
+
+
+class Coherent(Ket):
+    r"""The `N`-mode coherent state.
+
+    If ``x`` and/or ``y`` are iterables, their length must be equal to `1` or `N`. If their length is equal to `1`,
+    all the modes share the same parameters.
+
+    .. code-block::
+
+        >>> import numpy as np
+        >>> from mrmustard.lab_dev import Coherent, Vacuum, Dgate
+
+        >>> state = Coherent(modes=[0, 1, 2], x=[0.3, 0.4, 0.5], y=0.2)
+        >>> assert state.modes == [0, 1, 2]
+        >>> # assert Coherent(x=0.5, y=0.2) == Vacuum([0]) >> Dgate(x=0.5, y=0.2)
+
+    Args:
+        modes: The modes of the coherent state.
+        x: The `x` displacement of the coherent state.
+        y: The `y` displacement of the coherent state.
+        x_trainable: Whether the `x` displacement is trainable.
+        y_trainable: Whether the `y` displacement is trainable.
+        x_bounds: The bounds of the `x` displacement.
+        y_bounds: The bounds of the `y` displacement.
+
+    .. details::
+
+        For any :math:`\bar{\alpha} = \bar{x} + i\bar{y}` of length :math:`N`, the :math:`N`-mode
+        coherent state displaced :math:`N`-mode vacuum state is defined by
+
+        .. math::
+            V = \frac{\hbar}{2}I_N \text{and } r = \sqrt{2\hbar}[\text{Re}(\bar{\alpha}), \text{Im}(\bar{\alpha})].
+
+        Its ``(A,b,c)`` triple is given by
+
+        .. math::
+            A = O_{N\text{x}N}\text{, }b=\bar{\alpha}\text{, and }c=\text{exp}\big(-|\bar{\alpha}^2|/2\big).
+    """
+
+    def __init__(
+        self,
+        modes: Iterable[int],
+        x: Union[float, Iterable[float]] = 0.0,
+        y: Union[float, Iterable[float]] = 0.0,
+        x_trainable: bool = False,
+        y_trainable: bool = False,
+        x_bounds: Tuple[Optional[float], Optional[float]] = (None, None),
+        y_bounds: Tuple[Optional[float], Optional[float]] = (None, None),
+    ):
+        super().__init__("Coherent", modes=modes)
+        self._add_parameter(make_parameter(x_trainable, x, "x", x_bounds))
+        self._add_parameter(make_parameter(y_trainable, y, "y", y_bounds))
+
+    @property
+    def representation(self) -> Bargmann:
+        num_modes = len(self.modes)
+
+        xs = math.atleast_1d(self.x.value)
+        if len(xs) == 1:
```

Comment:
Do we also have this part of settings in triples.py as well?

Reply:
We do, but it's inevitable to add them here, due to the fact that we want the funcs in triples.py to be stand-alone.

Here, I have to make sure that `xs` and `ys` have the same length in all cases, including when `nmodes > 1` but `xs` and `ys` have length 1